File: Water Table Maximisation/ContainerWithWaterProblem.py

def maxArea(Area, Length) :
	area = 0
	for i in range(Length) :
		for j in range(i + 1, Length) :
		
			# Calculating the max area
			area = max(area, min(Area[j], Area[i]) * (j - i))
	return area

# Driver code
a = [ 1, 5, 4, 3 ]
b = [ 3, 1, 2, 4, 5 ]

len1 = len(a)
print(maxArea(a, len1))

len2 = len(b)
print(maxArea(b, len2))

File: Water Table Maximisation/ContainerWithWaterProblem.java

// Water Container 
import java.io.*;

class ContainerWithWaterProblem{

public static int maxArea(int[] a)
{

	int Area = 0;
	
	for(int i = 0; i < a.length; i++)
	{
		for(int j = i + 1; j < a.length; j++) 
		{
			Area = Math.max(
				Area, Math.min(a[i], a[j]) * 	(j - i));
		}
	}
	return Area;
}

// Driver code 
public static void main(String[] args)
{
	int a[] = { 10, 15, 6, 3 };
	int b[] = { 8, 7, 5, 3 };

	System.out.println(maxArea(a));
	System.out.println(maxArea(b));
}
}

// This code is contributed by mulchandanimanisha5
